User management for GKE (#29)
CircleCI-Public/server-terraform: gke/managing_user_access.md

  name: pod-reader
  namespace: develop
rules:
- apiGroups: [""] # "" indicates the core API group
  resources: ["pods"]
  verbs: ["list","get","watch"]
- apiGroups: ["extensions","apps"]
  resources: ["deployments"]
  verbs: ["get", "list", "watch"]
```

- and then apply the role to your cluster:
`kubectl apply -f read-role.yaml`

3. Now we will create a `role-binding` on the email address of our user's Google Cloud account and the pod-reader role in the cluster to define their access level.
- add the following to a file called `read-role-binding.yaml`
```
kind: RoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: pod-reader-binding
  namespace: develop
subjects:
- kind: User
  name: <user-email>
  apiGroup: rbac.authorization.k8s.io
roleRef:
  kind: Role
  name: pod-reader
  apiGroup: rbac.authorization.k8s.io
```

- and then apply the role-binding to your cluster:
`kubectl apply -f read-role-binding.yaml`

Now your user will be able to access the cluster from the bastion but will only be able to view pod resources in the develop namespace.

For more details on managing permissions, you may read Google's [RBAC] documentation.

## Other Role Binding Subjects

> Roles cannot only be bound to Google Cloud IAM users. The following other subjects are supported:

Reviewer: ```suggestion
RoleBindings are not limited to Google Cloud IAM users. The following other subjects are supported:
```
